### ISSUE TITLE
Remove use of $GLOBALS in Environment.php

### ISSUE THREAD

**Issue by emteknetnz** [silverstripe/silverstripe-framework#10717]
We use $GLOBALS in [Environment::getVariables()](https://github.com/silverstripe/silverstripe-framework/blob/4.12/src/Core/Environment.php#L43) and [Environment::setVariables()](https://github.com/silverstripe/silverstripe-framework/blob/4.12/src/Core/Environment.php#L60)

$GLOBALS is NOT all of the superglobals such as $_SERVER, instead it is all variables in the global namespace e.g. `global $myvar;`

According to [php.net](https://www.php.net/manual/en/reserved.variables.globals.php), in PHP 8.1, you can no longer write to $GLOBALS, which is what setVariables() [does](https://github.com/silverstripe/silverstripe-framework/blob/4.12/src/Core/Environment.php#L66)

$GLOBALS is probably a hangover from CMS 3 days where there was no .env file support and instead _ss_environment.php was used where global variables were writting in a PHP file

Since this is a breaking change (it still works if you're on PHP 7.4), We should retain $GLOBALS support in CMS4, and remove the use of it in CMS 5

## Acceptance criteria
- For CMS4, validates if going from PHP7.4 to PHP8.0/8.1 breaks this behaviour.
  - Update logic if we expect this to break user functionality.
  - Call it out in the doc if relevant.
- For CMS 5, removes any `$GLOBALS` references and call it out in the changelog.

## Note
- `getVariables` and `setVariables` do other things aside from messing around with $GLOBALS.

**Comment by GuySartorelli** [comment]
This is based on a false premise.
> According to [php.net](https://www.php.net/manual/en/reserved.variables.globals.php), in PHP 8.1, you can no longer write to $GLOBALS, which is what setVariables() [does](https://github.com/silverstripe/silverstripe-framework/blob/4.12/src/Core/Environment.php#L66)

This seems to be based on
> Warning As of PHP 8.1.0, write access to the entire $GLOBALS array is no longer supported

which then shows a series of examples of overriting _the entire variable_ i.e. `$GLOBALS = [];`

Further down, in another section about PHP 8.1, it has an explicit example of writing variables one-at-a-time _the exact same way `Environment` does_:

```php
// To restore the previous behavior, iterate its copy and assign each property back to $GLOBALS.
foreach ($globals as $key => $value) {
    $GLOBALS[$key] = $value;
}
```

I've validated that the behaviour in [Environment::getVariables()](https://github.com/silverstripe/silverstripe-framework/blob/4.12/src/Core/Environment.php#L43) and [Environment::setVariables()](https://github.com/silverstripe/silverstripe-framework/blob/4.12/src/Core/Environment.php#L60) work as expected in both CMS4 and CMS5 with php 8.1, and in CMS 5 with php 8.2

tl;dr nothing needs to change.
